modules/social_features/social_content_block/src/ContentBuilder.php

```diff
@@ -312,14 +312,15 @@ protected function sortBy(SelectInterface $query, EntityTypeInterface $entity_ty
       case 'most_commented':
         if ($entity_type_id === 'group') {
           $query->leftJoin('post__field_recipient_group', 'pfrg', "base_table.${entity_id_key} = pfrg.field_recipient_group_target_id");
-          $query->leftJoin('comment_field_data', 'cfd', "(base_table.${entity_id_key} = cfd.entity_id AND cfd.entity_type=:entity_type) OR (pfrg.entity_id = cfd.entity_id AND cfd.entity_type='post')", ['entity_type' => $entity_type_id]);
+          $query->leftJoin('group_content_field_data', 'gfd', "base_table.${entity_id_key} = gfd.gid");
+          $query->leftJoin('comment_field_data', 'cfd', "(base_table.${entity_id_key} = cfd.entity_id AND cfd.entity_type=:entity_type) OR (pfrg.entity_id = cfd.entity_id AND cfd.entity_type='post') OR (gfd.entity_id = cfd.entity_id AND cfd.entity_type='node')", ['entity_type' => $entity_type_id]);
         }
         // Otherwise only check direct votes.
-        elseif ($entity_type_id === 'node') {
+        else {
           $query->leftJoin('comment_field_data', 'cfd', "base_table.${entity_id_key} = cfd.entity_id AND cfd.entity_type=:entity_type", ['entity_type' => $entity_type_id]);
         }
 
-        $query->addExpression('COUNT(cfd.cid)', 'comment_count');
+        $query->addExpression('COUNT(distinct cfd.cid)', 'comment_count');
```

Comment:
@rollki Could you add why the `distinct` is needed here? I believe in the way we select the comments they should already be distinct, otherwise there's a different error.

`distinct` can cost quite a bit of performance so I'd prefer to avoid its use if we can.

Reply:
@Kingdutch 
We can not COUNT without `distinct` groups entities, because we have joined to `group_content_field_data` and `post__field_recipient_group` and this multiplies the sample. 
I could not solve this problem in a simple way, maybe with a subquery it is possible, but there will be more code, it will be unreadable and even slower.

But we can add `if` condition and don't do this for other entities.